src/consensus.rs

```diff
@@ -43,12 +38,12 @@ pub mod utils;
     Debug, Serialize, Deserialize, Clone, Encode, Decode, PartialEq, Eq, Hash, IntoStaticStr,
 )]
 pub enum ConsensusNetMessage {
-    StartNewSlot,
-    Prepare(ConsensusProposal),
+    Prepare(ConsensusProposal, Ticket),
     PrepareVote(ConsensusProposalHash),
-    Confirm(ConsensusProposalHash, QuorumCertificate),
+    // TODO: these should probably just be "SignedWithQuorumCertificate" to be consistent
+    Confirm(QuorumCertificate),
     ConfirmAck(ConsensusProposalHash),
-    Commit(ConsensusProposalHash, QuorumCertificate),
```

Comment:
I almost didn't see that change, but I think we can't remove it:

The quorum certificate is defined:

```rust
pub struct QuorumCertificate {
    signature: Signature,
    validators: Vec<ValidatorPublicKey>,
}
```

You don't know what it signs... So what do you trust with this signature ? That was the goal of the function `verify_commit_qc` that is removed and not replaced: re-build a message with the consensus proposal to check that the signature is really for the proposal we are handling. In the logic of this PR, it looks like we could accept a ConsensusProposal because we got a QuroumCertificate that looks valid, but is in fact for an other proposal...

Reply:
same for the confirm step & the removed function `verify_prepare_qc`

Reply:
Mh, la logique est que l'objet `message` qu'on passe à `verify_quorum_certificate` doit contenir l'objet en train d'être signé. Sauf si `BlstCrypto` ne fait pas ce que je pense (possible), pour moi ce que tu décris est impossible.

Je me trompe ?

Reply:
ah oui, pardon j'ai mal regardé, je dois pas avoir les trous en face des yeux Oo pardon XD

Reply:
mais du coup le message de "commit" arrive sans qu'on sache pour quel slot/view il est. On peut le déduire si on arrive en effet à reconstruire le message signé à partir de son propre état interne, mais c'est pas très explicite. On reçoit un Commit avec un certificat, et si la vérification de la signature est pas bonne, le node ne sait pas pourquoi : est-ce que le leader est byzantin et a signé un mauvais message ? ou c'est juste moi qui ait pas le bon proposal ?

Reply:
Oui je suis d'accord c'est moins explicite. C'est pour ça que dans un second temps je remettrai bien (suite à discussions avec @hhalex aussi le Slot et la View attendu explicitement sur tous les messages pour error de manière plus clean.

J'ai laissé ça pour plus tard parce que c'était en soit un changement facultatif, mais ça peut être fait avec.

Reply:
ok si c'est prévu en effet d'ajouter slot/view ça fait le taff